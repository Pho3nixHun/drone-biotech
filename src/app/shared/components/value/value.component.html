@let vM = vm();
<div *transloco="let t">
    @if (vM.type === 'withValueKey') {
        {{ t(vM.key, vM.params) }}
    } @else {
        {{ vM.value }}
    }
</div>
