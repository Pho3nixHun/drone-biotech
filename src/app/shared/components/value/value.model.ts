import { HashMap } from '@jsverse/transloco';

interface WithValueKey {
    type: 'withValueKey';
    key: string;
    params: HashMap;
}
interface WithoutValueKey {
    type: 'withoutValueKey';
    value: string | number | Date;
}

export type ValueVM = WithValueKey | WithoutValueKey;
