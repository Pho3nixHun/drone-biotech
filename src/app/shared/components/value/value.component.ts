import { Component, input } from '@angular/core';
import { ValueVM } from './value.model';
import { TranslocoModule } from '@jsverse/transloco';

@Component({
    selector: 'app-value',
    imports: [TranslocoModule],
    templateUrl: './value.component.html',
})
export class ValueComponent {
    public vm = input.required<ValueVM>();
}
