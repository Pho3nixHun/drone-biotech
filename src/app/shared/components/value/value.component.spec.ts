import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ValueComponent } from './value.component';
import { getTranslocoModule } from 'transloco-testing.module';
import { ValueVM } from './value.model';

describe('ValueComponent', () => {
    let fixture: ComponentFixture<ValueComponent>;
    let compiled: HTMLElement;
    beforeEach(async () => {
        await TestBed.configureTestingModule({
            imports: [
                ValueComponent,
                getTranslocoModule({
                    langs: { en: {} },
                    translocoConfig: {
                        availableLangs: ['en'],
                        defaultLang: 'en',
                    },
                }),
            ],
        }).compileComponents();

        fixture = TestBed.createComponent(ValueComponent);
        compiled = fixture.debugElement.nativeElement;
    });

    // Snapshot testing
    it('should render the template correctly if the vm does not contain valueKey', () => {
        // Arrange
        const valueVM: ValueVM = { type: 'withoutValueKey', value: 12 };
        fixture.componentRef.setInput('vm', valueVM);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
    // Snapshot testing
    it('should render the template correctly if the vm contains valueKey', () => {
        // Arrange
        const valueVM: ValueVM = {
            type: 'withValueKey',
            key: '',
            params: { count: 12 },
        };
        fixture.componentRef.setInput('vm', valueVM);

        // Act
        fixture.detectChanges();

        // Assert
        expect(compiled).toMatchSnapshot();
    });
});
