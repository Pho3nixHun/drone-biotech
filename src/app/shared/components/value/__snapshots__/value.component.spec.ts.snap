// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValueComponent should render the template correctly if the vm contains valueKey 1`] = `
<div
  id="root1"
  ng-version="19.2.9"
>
  <div>
      
    <!--container-->
    <!--container-->
  </div>
  <!--container-->
</div>
`;

exports[`ValueComponent should render the template correctly if the vm does not contain valueKey 1`] = `
<div
  id="root0"
  ng-version="19.2.9"
>
  <div>
    <!--container-->
     12 
    <!--container-->
  </div>
  <!--container-->
</div>
`;
